### PR TITLE
[RW-3201][risk=low] Re-attempt ubuntu dev docker

### PR DESCRIPTION
**Pull request by jmthibault79**
Resolves the stack trace crash due to incompatibility between Alpine and `opencensus-exporter-trace-stackdriver`.

### CODE REVIEW

File: api/docker-compose.yaml [hunk 1]
```diff
@@ -75,7 +75,7 @@ services:
     volumes:
       - ..:/w:cached
       - gradle-cache:/.gradle
-    entrypoint: ['with-uid.sh', 'wait-for', 'db:3306', --]
+    entrypoint: ['with-uid.sh', 'wait-for-it', 'db:3306', --]
```
Comment (jmthibault79) [inline api/docker-compose.yaml]: `wait-for` is a specifically Alpine tool

File: api/docker-compose.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ services:
       - ~/.config:/.config:cached
       - ~/.gsutil:/.gsutil:cached
   db:
-    image: mysql:5.7
+    image: mysql:5.7.27
```
Comment (jmthibault79) [inline api/docker-compose.yaml]: Not necessary.  But while I'm here...

File: api/src/dev/server/Dockerfile [hunk 3]
```diff
@@ -1,28 +1,39 @@
-FROM alpine:3.7
+FROM ubuntu:18.04
```
Comment (jmthibault79) [inline api/src/dev/server/Dockerfile]: The changes in this file are the same as in #2471 except for `wait-for`/`wait-for-it`

File: api/src/dev/server/Dockerfile [hunk 4]
```diff
@@ -1,28 +1,39 @@
...
+     git=1:2.17.1-1ubuntu0.4 \
+     gettext=0.19.8.1-6ubuntu0.3 \
+     mysql-server=5.7.27-0ubuntu0.18.04.1 \
+     mysql-client=5.7.27-0ubuntu0.18.04.1 \
+     ruby=1:2.5.1 \
+     ruby-dev=1:2.5.1 \
+     ruby-json=2.1.0+dfsg-2 \
+     make=4.1-9.1ubuntu1 \
+     unzip=6.0-21ubuntu1 \
+     wait-for-it=0.0~git20170723-1 \
```
Comment (jmthibault79) [inline api/src/dev/server/Dockerfile]: Wait for a specific port to be accessible, ubuntu version

File: api/src/dev/server/Dockerfile [hunk 5]
```diff
@@ -36,16 +47,9 @@ RUN apk --no-cache add \
...
 RUN gcloud components install app-engine-java
 
-RUN apk --no-cache add gettext
-
-RUN apk --no-cache add mysql
-RUN apk --no-cache add mysql-client
 RUN curl https://dl.google.com/cloudsql/cloud_sql_proxy.linux.amd64 \
   > /usr/local/bin/cloud_sql_proxy && chmod +x /usr/local/bin/cloud_sql_proxy
 
-RUN curl https://raw.githubusercontent.com/mrako/wait-for/d9699cb9fe8a4622f05c4ee32adf2fd93239d005/wait-for \
```
Comment (jmthibault79) [inline api/src/dev/server/Dockerfile]: Wait for a specific port to be accessible, alpine version

**Comment by jmthibault79** [comment]
Requesting review from a Mac user and a Linux user.  Please verify via the following method or equivalent:

* `./project.rb docker-clean` 
* `./gradlew clean compileGeneratedJava` 
* `./project.rb run-local-all-migrations`
* `./project.rb dev-up`
* Verify local API is working
* Stop local API
* `./project.rb start-api-reqs`
* `./gradlew appengineRun`
* Verify local API is working (again)

File: api/docker-compose.yaml (hunk 1 above)
Comment (jmthibault79) [reply api/docker-compose.yaml]: Running `docker-compose` with `wait-for` entrypoints fails with `Operation timed out`.

**Comment by blrubenstein** [comment]
@jaycarlton I believe uses a Mac, so we should have @calbach check the Linux run.

**Comment by calbach** [comment]
Trying this now.

Out of curiosity, why `docker system prune -a --volumes` in the example commands? Just out of caution? `docker-clean` should dispose of any project-relevant volumes AFAIK.

**Comment by jmthibault79** [comment]
@calbach paranoia mostly

**Comment by blrubenstein** [comment]
I think I had some stale docker images that for some reason didn't get cleaned up. But may have been from something else.

**Comment by jmthibault79** [comment]
Thanks!

**Pull request merged**